accord-core/src/main/java/accord/local/Commands.java

```diff
@@ -610,247 +665,236 @@ private static boolean maybeExecute(SafeCommandStore safeStore, SafeCommand safe
         }
     }
 
-    protected static WaitingOn populateWaitingOn(SafeCommandStore safeStore, TxnId txnId, Timestamp executeAt, PartialDeps partialDeps)
+    protected static WaitingOn initialiseWaitingOn(SafeCommandStore safeStore, TxnId waitingId, Timestamp executeWaitingAt, PartialDeps partialDeps, Route<?> route)
     {
-        Ranges ranges = applyRanges(safeStore, executeAt);
-        if (ranges.isEmpty())
-            return WaitingOn.EMPTY;
-
-        return populateWaitingOn(safeStore, ranges, txnId, executeAt, partialDeps);
+        Unseekables<?> executionParticipants = route.participants().slice(safeStore.ranges().allAt(executeWaitingAt));
+        WaitingOn.Update update = new WaitingOn.Update(executionParticipants, partialDeps);
```

Comment:
Ah, yes. I think this was because the conditional breakpoint was slowing things down too much. This is fine, and correct if there are no execution participants. I will delete this.